Sushi.Mediakiwi.Data.Elastic/Repositories/NotificationRepository.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Nest;
+using Sushi.Mediakiwi.Data.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Sushi.Mediakiwi.Data.Elastic.Repositories
+{
+    /// <summary>
+    /// Provides methods to store and retrieve instances of <see cref="Notification"/> from an Elastic server. 
+    /// It is advised to use one instance of <see cref="NotificationRepository"/> per application or at least per Elastic connection.
+    /// </summary>
+    public class NotificationRepository : INotificationRepository
+    {
+        private Nest.IElasticClient _client;
+        private string _dataStreamName;
+        public NotificationRepository(Nest.IElasticClient client, string dataStreamName)
+        {
+            // create client
+            _client = client;
+            _dataStreamName = dataStreamName;
+        }
+        
+        public Data.Notification Save(Data.Notification notification)
+        {
+            var elasticNotification = new Notification(notification);
+            var response = _client.Index(elasticNotification, i => i.Index(_dataStreamName));
+            elasticNotification.ElasticId = new ElasticId(response.Id, response.Index);
+            return elasticNotification;
+        }
+
+        public async Task<Data.Notification> SaveAsync(Data.Notification notification)
+        {
+            var elasticNotification = new Notification(notification);
+            var response = await _client.IndexAsync(elasticNotification, i => i.Index(_dataStreamName));
+            elasticNotification.ElasticId = new ElasticId(response.Id, response.Index);
+            return elasticNotification;
+        }
+
+        public async Task<List<NotificationGroup>> GetAggregatedGroupsAsync(NotificationType? selectionID, DateTime? from, DateTime? to)
+        {
+            var response = await _client.SearchAsync<Notification>(s => s
+                .Size(0)
+                .Index(_dataStreamName)
+                .Query(q =>
+                    +q.Term(t => t.Selection, selectionID)
+                    && +q.DateRange(d => d.GreaterThanOrEquals(from))
+                    && +q.DateRange(d => d.LessThan(to))
```

Comment:
The from date is inclusive, the to date is exclusive, that should probably be described in the params

Reply:
its pretty standard behavior for date ranges to have the upper bound exclusive and the lower bound inclusive (give me everything between 2020-01-01 and 2021-02-1)

Reply:
Agreed, but still noteworthy